### DISCUSSION TITLE
Filtering Journal Entries by Location

### DISCUSSION THREAD

**Discussion by jeffmacdonald** [netbox-community/netbox#10753] [Ideas]
Hi!

At my work we have ~10 datacenters around the world. When techs arrive they'd like to be able to see a list of servers that we've identified as having issues via the Journal. Currently they just see a giant list.

Thoughts?

**Comment by mtinberg** [comment]
Journal entries are searchable by Tags, Kind, timeframe and a text Search, maybe custom_fields as well, so it'd make sense to use one of those features.  For example, you could have 10 tags for the datacenters and tag journal entries with the datacenter (and any other attributes you might want to search on) to make them easily filterable, failing that you could try and consistently use the DC name in the text of the journal and search for that, but that seems more error prone.


—

Mark Tinberg ***@***.***>

Division of Information Technology-Network Services

University of Wisconsin-Madison


________________________________
From: Jeff MacDonald ***@***.***>
Sent: Wednesday, October 26, 2022 8:09 AM
To: netbox-community/netbox ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [netbox-community/netbox] Filtering Journal Entries by Location (Discussion #10753)


Hi!

At my work we have ~10 datacenters around the world. When techs arrive they'd like to be able to see a list of servers that we've identified as having issues via the Journal. Currently they just see a giant list.

Thoughts?

**Reply by jeffmacdonald** [reply]
Sadly both feel error prone when the meta data of "Location" is already attached to the device, and journal entries are attached to devices. Seems like the "data entry" half of this is automatic and doesn't need to happen, which is better for users in the end.

**Comment by mtinberg** [comment]
replying to myself,

another idea is you could use an API client script or manually put in the search parameters, /api/docs says that assigned_object_type_id and assigned_object_id are searchable fields so you should be able to make a Netbox Report that organizes journal entries by site by looking up all the devices at that site, or an external script that plugs all the device id numbers into assigned_object_id parameters

eg.

/extras/journal-entries/?assigned_object_type_id=21&assigned_object_id=66

you should be able to just keep adding additional assigned_object_id parameters with the list of devices at that site you want to search for, up to the URL limit you've configured in your web service.


—

Mark Tinberg ***@***.***>

Division of Information Technology-Network Services

University of Wisconsin-Madison


________________________________

**Comment by ZPrimed** [comment]
This almost feels like it would be a report of some sort?
My org could also benefit from the ability to search Journal entries by Site and/or Location though. We use site as "physical building," and then Locations within can be any MDF/IDF closet, or masts/poles on rooftop (work is a WISP).
